src/org/jitsi/impl/neomedia/rtp/RTPEncodingImpl.java

```diff
@@ -279,11 +279,15 @@ public boolean matches(byte[] buf, int off, int len)
             return false;
         }
 
+        if (temporalId == -1)
+        {
+            return true;
+        }
+
         int tid = track.getMediaStreamTrackReceiver()
             .getStream().getTemporalID(buf, off, len);
 
-        return (tid == -1 && ArrayUtils.isNullOrEmpty(dependencyEncodings))
-                || tid == temporalId;
+        return tid == temporalId;
```

Comment:
Maybe you're right but my concern here is that we're trying to not break something that we don't know how it works to begin with. If we're doing simulcast, I expect I expect to see positive TIDs. If that doesn't hold anymore, I prefer to see warnings or exceptions than have something that miraculously works most of the time, but then it mysteriously fails some times. In other words, unexpected behavior is interesting and needs to be investigated and understood, not masked.